tests/test_modern.py

```diff
@@ -740,3 +755,87 @@ def test_ignore_with_debris_cleanup():
         assert not (directory / '.cache').exists()
         assert (directory / 'foo' / '.cache').exists()
         assert (directory / 'foo' / '.cache' / 'test.txt').exists()
+
+
+def test_debris_cleanup_scans_directories_once():
+    """
+    Does debris cleanup scan each directory only once per topic,
+    rather than once per glob pattern?
+    """
+    with TemporaryDirectory() as tmp:
+        directory = Path(tmp)
+
+        # Create a directory structure with ignored directories
+        (directory / '.git').mkdir()
+        (directory / 'subdir1').mkdir()
+        (directory / 'subdir2').mkdir()
+
+        # Mock the should_ignore function to count calls
+        original_should_ignore = pyclean.modern.should_ignore
+        call_count = {'total': 0, 'git_checks': 0}
+
+        def counting_should_ignore(path, patterns):
+            call_count['total'] += 1
+            if path.name == '.git':
+                call_count['git_checks'] += 1
+            return original_should_ignore(path, patterns)
+
+        args = Namespace(dry_run=False, ignore=['.git'])
+        pyclean.modern.Runner.configure(args)
+
+        with patch('pyclean.modern.should_ignore', side_effect=counting_should_ignore):
+            # Remove debris for a topic with multiple patterns ('cache' has 2 patterns)
+            pyclean.modern.remove_debris_for('cache', directory)
+
+        # Each subdirectory should be checked once per directory traversal,
+        # not once per glob pattern
+        # With 3 subdirs (.git, subdir1, subdir2), we expect:
+        # - 1 check for .git at root level
+        # - 1 check for .git inside subdir1
+        # - 1 check for .git inside subdir2
+        # Total: at most 3 checks for .git (not 6 if it were called for each pattern)
+        assert call_count['git_checks'] == 1, (
+            f'Expected 1 check for .git directory, but got {call_count["git_checks"]}'
+        )
+
+
+@patch('pyclean.modern.log')
+@patch('os.scandir', side_effect=PermissionError('Permission denied'))
+def test_recursive_delete_debris_logs_permission_error(mock_scandir, mock_log):
+    """
+    Does recursive_delete_debris log a warning when directory cannot be accessed?
+    """
+    args = Namespace(dry_run=False, ignore=[])
+    pyclean.modern.Runner.configure(args)
+
+    directory = Path('/tmp/test')
+    patterns = ['.cache/**/*', '.cache/']
+
+    pyclean.modern.recursive_delete_debris(directory, patterns)
+
+    # Check that warning was logged
+    mock_log.warning.assert_called_once()
+    call_args = mock_log.warning.call_args[0]
+    assert 'Cannot access directory' in call_args[0]
```

Comment:
What is `call_args[0]`? The array-notation access makes the code unclear. Is there a better way to write this?

If possible, use a single line to accomodate the current two lines.